src/Collections/Sequences/Seq.dfy

```diff
@@ -803,6 +827,26 @@ module {:options "-functionSyntax:4"} Seq {
     }
   }
 
+  /* Optimized implementation of Flatten(Map(f, xs)). */
+  function {:opaque} FlatMap<T,R>(f: (T ~> seq<R>), xs: seq<T>): (result: seq<R>)
+    requires forall i :: 0 <= i < |xs| ==> f.requires(xs[i])
+    ensures result == Flatten(Map(f, xs));
```

Comment:
Good point! I'm not sure in this particular case whether it would be better to keep it opaque and move the `result == Flatten(Map(f, xs))` to a lemma, so I'll take your suggestion as the easier option at least.